### ISSUE TITLE
Is it possible to bridge two kafka clusters with this?

### ISSUE THREAD

**Issue by josdotso** [edenhill/kafkacat#132]
Such that message posting to topic1 at clusterA arrives also at topic1 at clusterB...?

**Comment by edenhill** [comment]
Yes, you can pipe a kafkacat consumer for cluster A to a kafkacat producer for cluster B.
But do note that only key and value will be persisted, not message timestamps (will be fixed, see #125) or offsets (not fixable).

**Comment by jeffwidman** [comment]
If the topic names are the same, why wouldn't you use mirrormaker for this?

**Comment by JeanMertz** [comment]
> But do note that only key and value will be persisted, not message timestamps

I wasn't looking for the same functionality as requested in this issue, but I _was_ looking for ways to set the timestamp of a produced message.

My use-case is as such:

I have multiple (large) files with messages separated by a newline, I produced those messages using `-P -l`, but realised the timestamp was set to the commit timestamp, while I have a timestamp embedded in the (JSON-formatted) messages that I'd like to use as the Kafka message timestamp.

So instead, using `jq`, I now have reformatted the lines in the files to contain `<RFC3999 timestamp>|||<JSON message body>`. I'd like to supply this file to kafkacat to efficiently produce all the messages in this file, but right now this is not supported, and I've had to write a small go utility to do the job.

Any chance of supporting such a use-case?

**Comment by edenhill** [comment]
There's been some discussion on supporting JSON input to the producer, which would allow more specific per-message attributes to be set.
